waspc/data/Generator/templates/react-app/src/test/vitest/helpers.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import { BrowserRouter as Router } from 'react-router-dom'
+import { render, RenderResult, cleanup } from '@testing-library/react'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { beforeAll, afterEach, afterAll } from 'vitest'
+import { Query } from '../../queries'
+import config from '../../config'
+import { type HttpMethod } from '../../types'
+
+export function renderInContext(ui: React.ReactElement): RenderResult {
+  const client = new QueryClient()
+  const { rerender, ...result } = render(
+    <QueryClientProvider client={client}><Router>{ui}</Router></QueryClientProvider>
+  )
+  return {
+    ...result,
+    rerender: (rerenderUi: React.ReactElement) =>
+      rerender(
+        <QueryClientProvider client={client}><Router>{rerenderUi}</Router></QueryClientProvider>
+      )
+  }
+}
+
+type QueryRoute = Query<any, any>['route']
+type Handler = () => void
+
+export function mockServer() {
+  const server = setupServer()
+
+  beforeAll(() => server.listen())
+  afterEach(() => {
+    server.resetHandlers()
+    cleanup()
+  })
+  afterAll(() => server.close())
+
+  function mockQuery({ route }: { route: QueryRoute }, resJson: any): void {
+    const url = `${config.apiUrl}${route.path}`
+    const responseHandler = (_req, res, ctx) => {
```

Comment:
We're not using `strict: true`, in our TS config because we're trying to match the TS config we generate for our users.

Still, for best development experience, I recommend setting it to `true` while working. It's going to help you catch many additional type errors. 

No type errors here, btw. I just noticed it was off thanks to the implicit `any`. If the function was longer, it would be a good idea to do something like this and get proper type support inside:
```typescript
const responseHandler: ResponseResolver<any, RestContext, any> = (_req, res, ctx) => {
  return res(ctx.json(resJson))
}
```